### PR TITLE
[WIP] Add tool for starting an ODFE node on a desktop

### PR DESCRIPTION
**Pull request by camerski**
This tool is intended to make it easier for developers to set up a complete ODFE release candidate on a desktop so that test failures can be debugged. Right now it only supports a tarball install, but eventually it will support installing from deb and rpm (using Docker) as well.

*Test Results:*
Just some manual testing. It seems to work.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: release-tools/desktop-cluster/start-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    echo "-w\tSet the working directory. This directory will be deleted and recreated. If not specified, then a new random temporary directory will be used. The tarballs will be unpacked into this directory; logs and the process pid files will be stored here."
+    echo "-a\tSet the maximum number of times to probe the elasticsearch and kibana processes to see if the services are up. Defaults to 12."
+    echo "-b\tSet the backoff (in seconds) between probes to the elasticsearch and kibana processes to see if they are up. Defaults to 10."
+    echo "-h\tPrint this help message."
+}
+
+MAX_ATTEMPTS=12
+BACKOFF=10
+
+while getopts ":he:k:w:uv:a:b:" arg; do
```
Comment (peterzhuamazon) [inline release-tools/desktop-cluster/start-cluster.sh:25]:
```
:he:k:w:uv:a:b: or :h:e:k:w:u:v:a:b:
```

Comment (camerski) [reply release-tools/desktop-cluster/start-cluster.sh:25]: The -h and -u options do not require arguments.

Comment (peterzhuamazon) [reply release-tools/desktop-cluster/start-cluster.sh:25]: Forgot about that. Thanks.

File: release-tools/desktop-cluster/stop-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+if [ -z "$1" ]; then
+    echo "Usage: $0 /path/to/workdir"
+fi
+DIR=$1
+
+if [ -e $DIR/elasticsearch.pid ]; then
+    pid=$(<$DIR/elasticsearch.pid)
+    echo "Killing Elasticsearch (pid $pid)"
+    kill $pid
+    rm $DIR/elasticsearch.pid
```
Comment (gaiksaya) [inline release-tools/desktop-cluster/stop-cluster.sh:15]: I don't see removing the $WORK_DIR anywhere after killing the processes (stopping the clusters). If everytime a new directory is created, should we add a step to remove the earlier one? Not sure if it is necessary as it is anyway created in /tmp

Comment (camerski) [reply release-tools/desktop-cluster/stop-cluster.sh:15]: Right now the log files are stored in $WORK_DIR, so I don't want to delete it automatically. Eventually this should change - we should be able to specify a log directory and have it automatically delete $WORK_DIR so you don't wind up with gigabytes of files in /tmp. That's a future enhancement.

**Comment by peterzhuamazon** [comment]
Close this as we already have a new deployment script in opensearch-project/opensearch-build/pull/64. there.

**Pull request closed without merging**